app/components/application/AppEndpointUrl/AppEndpointUrl.tsx

```diff
@@ -12,19 +19,57 @@ export const links = () => {
 
 type AppEndpointUrlProp = InputProps & {
   chain: Blockchain | undefined | null
+  handleRemove: () => void
+  hasDelete: boolean
 }
 
-export default function AppEndpointUrl({ chain, ...props }: AppEndpointUrlProp) {
+export default function AppEndpointUrl({
+  chain,
+  handleRemove,
+  hasDelete,
+  ...props
+}: AppEndpointUrlProp) {
+  const theme = useMantineTheme()
+
   if (!chain) {
     return <></>
   }
 
   return (
     <div className="pokt-app-endpoint-url">
-      <div className="pokt-app-endpoint-url-abbrv">
-        <ChainWithImage chain={chain.description} label={chain.ticker} />
-      </div>
-      <TextInput {...props} />
+      <Badge
```

Comment:
This is looking better, but im not 100% sure about how round it is. We dont really use that much border-radius anywhere else. Ill check with gabi to see what she thinks.